soc/xtensa/espressif_esp32/esp32s3/Kconfig.series

```diff
@@ -12,5 +12,17 @@ config SOC_SERIES_ESP32S3
 	select XIP if !MCUBOOT
 	select HAS_ESPRESSIF_HAL
 	select CPU_HAS_FPU
+
+config SOC_ESP32S3_PROCPU
+	bool "Application runs in ESP32S3 PROCPU (core 0)"
```

Comment:
This looks like it should be enabled in `esp32s3_devkitm_defconfig`, to indicate the default core used by the board.

However, I see that this also enables starting APPCPU, and that should be made into a separate config first.

Reply:
@57300, as you noted, I used `CONFIG_SOC_ESP32S3_PROCPU` as a mechanism to indicate that seconds core is enabled and should be started. It should be set enabled by default at `esp32s3_devkitm_defconfig` unless AMP scenario (`ipc_samples`) is used. Sounds ok?

Reply:
I understand, but adding Kconfigs for `SOC_ESP32S3_PROCPU` and `SOC_ESP32S3_APPCPU` suggests a pattern that other AMP SoCs in Zephyr already use. If you look at similar examples:
* `SOC_MIMXRT1166_CM4` and `SOC_MIMXRT1166_CM7`
* `SOC_MPS2_AN521_CPU0` and `SOC_MPS2_AN521_CPU1`
* `SOC_NRF5340_CPUAPP_QKAA` and `SOC_NRF5340_CPUNET_QKAA`

those symbols are used to determine which CPU that you are building for, and they are enabled in board defconfigs. In addition, the SoCs in question have separate configs for starting the second core, like `SECOND_CORE_MCUX` or `BOARD_ENABLE_CPUNET`.

With that in mind, my point is that `SOC_ESP32S3_PROCPU` is a bit misleading, especially given its prompt:
> Application runs in ESP32S3 PROCPU (core 0)

I'm suggesting to add another config with a different prompt, which specifically relates to dual core operation.

Reply:
> those symbols are used to determine which CPU that you are building for, and they are enabled in board defconfigs. In addition, the SoCs in question have separate configs for starting the second core, like `SECOND_CORE_MCUX` or `BOARD_ENABLE_CPUNET`.

I don't agree with that. The SOC cores amongst vendors have miscellaneous names. The ESP32S3 cores are very clearly named `PROCPU` (Core0) and `APPCPU` (Core1).

Reply:
Just an extra point, there is no way to make APPCPU as default core regardless the board configuration, we should remember que AMP on ESP32 or S3 are a soft-implementation, they're actually symmetric-processing units where the APPCPU is never allowed to run without PROCPU

Reply:
@57300, I see your point. I had a discussion about this prior this PR exactly about having some "ENABLE_APPCPU" extra configuration. Our conclusion was that, as AMP support has not yet some common and defined structure among vendors, we decided to only include those options (`PROCPU` and `APPCPU`) instead having that additional config.  

`PROCPU` "help" clearly says that it automatically triggers APPCPU when it is set:  

```c
config SOC_ESP32S3_PROCPU
	bool "Application runs in ESP32S3 PROCPU (core 0)"
	depends on SOC_SERIES_ESP32S3
	help
	  When this SOC is enabled, it will run application on PROCPU (core 0). It will automatically
	  enables AMP support by building, flashing and loading APPCPU (core 1) image if exists.
```
Based on prior comments and on this, do you still need changes in this PR to accept as is?

Reply:
> AMP support has not yet some common and defined structure among vendors

@sylvioalves I think you're right to some extent, certainly about the "start second core" configuration - some products have it, some don't, and there's no consistent naming. For core selection, there is the more prevalent pattern of placing e.g., `PROCPU` and `APPCPU` into a `choice`, but there are definite outliers today. Plus, SoC configuration is always subject to change with upcoming work anyway, including https://github.com/zephyrproject-rtos/zephyr/issues/43488.

While the help text for `SOC_ESP32S3_PROCPU` looks fine, I would still recommend at least renaming it to `SOC_ESP32S3_ENABLE_APPCPU`, to reflect that this config only has to do with AMP support. You can't enable `SOC_ESP32S3_PROCPU` when building a single-core app for `PROCPU`, because it will fail to link. Instead, it's practically `SOC_ESP32S3_APPCPU` that tells you whether you're building for `APPCPU` or `PROCPU`.

But this can be improved in a future PR. I will dismiss my -1.